pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -423,6 +430,40 @@ func (r *ReconcileEgressGateway) reconcileEgressGateway(ctx context.Context, egw
 	return nil
 }
 
+func (r *ReconcileEgressGateway) getNetworkingPullSecrets(pullSecrets []*v1.Secret, ns string) ([]*v1.Secret, error) {
```

Comment:
We should pass a context into this function and use it, rather than using `context.Background()`